packages/xo-server-transport-icinga2/src/index.js

```diff
@@ -0,0 +1,131 @@
+import assert from 'assert'
+import createLogger from '@xen-orchestra/log'
+import httpRequestPlus from 'http-request-plus'
+import { format } from 'url'
+
+// =============================================================================
+
+const log = createLogger('xo:xo-server:transport-icinga2')
+
+// =============================================================================
+
+export const configurationSchema = {
+  type: 'object',
+
+  properties: {
+    server: {
+      type: 'string',
+      description: 'The icinga2 server http/https address',
+    },
+    port: {
+      type: 'integer',
+      description: 'The icinga2 server access port',
+      default: 5665, // Default icinga2 port
+    },
+    user: {
+      type: 'string',
+      description: 'The icinga2 server username',
+    },
+    password: {
+      type: 'string',
+      description: 'The icinga2 server password',
+    },
+    filter: {
+      type: 'string',
+      description: 'The filter to use',
+    },
+    acceptUnauthorized: {
+      type: 'boolean',
+      description: 'Accept unauthorized certificates',
+      default: false,
+    },
+  },
+  additionalProperties: false,
+  required: ['server', 'user', 'password', 'filter'],
+}
+
+// =============================================================================
+
+export const STATUS_MAP = {
+  OK: 0,
+  WARNING: 1,
+  CRITICAL: 2,
+  UNKNOWN: 3,
+}
+
+// =============================================================================
+
+class XoServerIcinga2 {
+  constructor({ xo }) {
+    this._xo = xo
+  }
+
+  // ---------------------------------------------------------------------------
+
+  configure(configuration) {
+    const ENDPOINT = '/v1/actions/process-check-result'
+    const split = configuration.server.split('://')
```

Comment:
Bad idea to parse an URL by hand, you can use [`URL`](https://devdocs.io/node~8_lts/url#url_class_url) and it's setters to construct the final URL.